sys/openbsd/init.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2017 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package openbsd
+
+import (
+	"github.com/google/syzkaller/prog"
+	"github.com/google/syzkaller/sys/targets"
+)
+
+type arch struct {
+	MAP_ANONYMOUS uint64
+	MAP_PRIVATE   uint64
+	MAP_FIXED     uint64
+	PROT_READ     uint64
+	PROT_WRITE    uint64
+	S_IFMT        uint64
+	S_IFCHR       uint64
+	S_IFBLK       uint64
+	S_IFIFO       uint64
+}
+
+func InitTarget(target *prog.Target) {
+	arch := &arch{
+		MAP_ANONYMOUS: target.ConstMap["MAP_ANON"],
+		MAP_FIXED:     target.ConstMap["MAP_FIXED"],
+		MAP_PRIVATE:   target.ConstMap["MAP_PRIVATE"],
+		PROT_READ:     target.ConstMap["PROT_READ"],
+		PROT_WRITE:    target.ConstMap["PROT_WRITE"],
+		S_IFBLK:       target.ConstMap["S_IFBLK"],
+		S_IFCHR:       target.ConstMap["S_IFCHR"],
+		S_IFIFO:       target.ConstMap["S_IFIFO"],
+		S_IFMT:        target.ConstMap["S_IFMT"],
+	}
+
+	target.MakeMmap = targets.MakePosixMmap(target)
+	target.SanitizeCall = arch.sanitizeCall
+}
+
+func (arch *arch) sanitizeCall(c *prog.Call) {
+	mknodModeIdx := -1
+	switch c.Meta.CallName {
+	case "mmap":
+		// Add MAP_FIXED flag, otherwise it produces non-deterministic results.
+		c.Args[3].(*prog.ConstArg).Val |= arch.MAP_FIXED
```

Comment:
We have a common helper for this function that is used on linux/netbsd/freebsd. netbsd/freebsd have 0 custom code here now. Can you just use that function here? Not sure if you are doing something more complex with mknod here. If yes, could we improve the common function to do it too, and then use the common function?

Reply:
I did a compromise and started using the helper function but kept the `VBAD` parts since it removes unwanted crashes.